### DISCUSSION TITLE
How Can I use `sliding_window_inference` for a network that has two branches outputting different things?

### DISCUSSION THREAD

**Discussion by OeslleLucena** [Project-MONAI/MONAI#1499] [Q&A]
Hi, 

I am currently having the following error and I would appreciate how I could sort that out.

```
  File "/home/ol18/miniconda3/envs/tract_uncertainty/lib/python3.6/site-packages/pytorch_lightning/accelerators/gpu_accelerator.py", line 66, in _step
    output = model_step(*args)
  File "/home/ol18/Codes/TractUncertainty/segmentation_sh_torchIO_PIL_MONAI.py", line 281, in validation_step
    outputs = sliding_window_inference(inputs, roi_size, sw_batch_size, self.forward)
  File "/home/ol18/miniconda3/envs/tract_uncertainty/lib/python3.6/site-packages/monai/inferers/utils.py", line 129, in sliding_window_inference
    seg_prob = predictor(window_data, *args, **kwargs).to(device)  # batched patch segmentation
AttributeError: 'tuple' object has no attribute 'to'
```

Many thanks!

**Comment by rijobro** [comment]
The output of `predictor` should be of type `torch.Tensor` so that it can be copied to the device (e.g., GPU) with `.to(device)`. For whatever reason, in your case `predictor` is returning a `tuple` (perhaps multiple `torch.Tensor`?).

I can't say much else without seeing more code, but I think you need to put in some print statements and check that the return types are as you expect throughout the script.

**Reply by OeslleLucena** [reply]
@rijobro  outputs from `predictor` was indeed a `tuple`. I have adapted the script of `sliding_window_inference` to handle two outputs instead of one suiting my needs. Thanks for the help.

**Reply by CataRay** [reply]
@OeslleLucena could you share your changes to `sliding_window_inference`? I am running into a similar problem when trying to get the attention maps out of my network. [my question](https://github.com/Project-MONAI/MONAI/discussions/4522)
